src/main/java/de/retest/web/DefaultValuesProvider.java

```diff
@@ -0,0 +1,95 @@
+package de.retest.web;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import de.retest.ui.DefaultValueFinder;
+import de.retest.ui.descriptors.IdentifyingAttributes;
+
+public class DefaultValuesProvider implements DefaultValueFinder {
+
+	public static final class ReadDefaultValuesException extends RuntimeException {
+
+		private static final long serialVersionUID = 1L;
+
+		public ReadDefaultValuesException( final Exception e ) {
+			super( "Could not load default CSS values from " + DEFAULTS_FILE_PATH, e );
+		}
+	}
+
+	public static final String DEFAULTS_FILE_PATH = "/defaults.yaml";
+
+	private final Map<String, Map<String, String>> defaultValues;
+
+	public DefaultValuesProvider() {
+		try ( final InputStream url = getClass().getResourceAsStream( DEFAULTS_FILE_PATH ) ) {
+			defaultValues = readAttributesConfigFromFile( url );
+		} catch ( final Exception e ) {
+			throw new ReadDefaultValuesException( e );
+		}
+	}
+
+	private Map<String, Map<String, String>> readAttributesConfigFromFile( final InputStream in ) throws IOException {
+		final Map<String, Map<String, String>> defaultValues = new HashMap<>();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename "defaultValues" which hides the field declared at line 32. ([squid:HiddenFieldCheck](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AHiddenFieldCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=52&issues=AWURSaL-yxJLWg29qAp0&open=AWURSaL-yxJLWg29qAp0)

[ISSUE_KEY]: AWURSaL-yxJLWg29qAp0